tests/ttnn/unit_tests/operations/test_moreh_adam.py

```diff
@@ -141,3 +144,22 @@ def forward(self, x):
         logger.debug(f"Out passing (max_exp_avg_sq)={passing}")
         logger.debug(f"Output pcc={out}")
     assert passing
+
+
+@pytest.mark.parametrize(
+    "params",
+    (
+        # shape, lr, betas, eps, weight_decay, amsgrad, fp32_dest_acc_en
+        ([32, 32], 0.0, (0.9, 0.999), 1e-06, 0.0, True, True),
+        ([2, 2, 2, 2, 2, 2, 64, 64], 0.0, (0.9, 0.999), 1e-06, 0.0, False, False),
+    ),
+)
+def test_moreh_adam_enable_cache(params, device, use_program_cache):
+    for i in range(4):
+        if i % 2 == 1:
+            amsgrad = not amsgrad
+
+        shape, lr, betas, eps, weight_decay, amsgrad, fp32_dest_acc_en = params
+        test_moreh_adam(shape, lr, betas, eps, weight_decay, amsgrad, fp32_dest_acc_en, device)
```

Comment:
```suggestion
    for i in range(2):
        shape, lr, betas, eps, weight_decay, amsgrad, fp32_dest_acc_en = params
        test_moreh_adam(shape, lr, betas, eps, weight_decay, amsgrad, fp32_dest_acc_en, device)
```
In the above code, when `i%2 == 1`, executing `amsgrad = not amsgrad` seems to have no effect because the value of amsgrad is overwritten right below it.

Reply:
I've changed the order of the operations, should work now.